src/wallet/coin_selection.rs

```diff
@@ -261,7 +296,28 @@ impl<D: Database> CoinSelectionAlgorithm<D> for OldestFirstCoinSelection {
                 .chain(optional_utxos.into_iter().map(|utxo| (false, utxo)))
         };
 
-        select_sorted_utxos(utxos, fee_rate, amount_needed, fee_amount)
+        select_sorted_utxos(utxos, fee_rate, amount_needed, fee_amount, drain_script)
+    }
+}
+
+fn decide_change(remaining_amount: u64, fee_rate: FeeRate, drain_script: &Script) -> Excess {
+    // drain_output_len = size(len(script_pubkey)) + len(script_pubkey) + size(output_value)
```

Comment:
From the comment for the `drain_output_len` computation, I was expecting, a third component in the form of the size of the `script_pubkey` in the computation below. I only see the addition of two components. Is there something, I'm missing?

Reply:
When you `serialize` a `Script` rust-bitcoin prefixes it with the length (which is a VarInt), so the first term amounts for both `serialize(len(script_pubkey)) + len(script_pubkey)`.

As I explained in the other comment, using `Script::len()` would have only returned the length of the script itself, without accounting for the VarInt preceeding it.

But the way this is done should be correct.

Also check out: https://github.com/bitcoindevkit/bdk/pull/630#discussion_r906696742